src/zap/zaprelocs.cpp

```diff
@@ -112,12 +112,12 @@ void ZapBaseRelocs::WriteReloc(PVOID pSrc, int offset, ZapNode * pTarget, int ta
         {
             TADDR pSite = (TADDR)m_pImage->GetBaseAddress() + rva;
 
-            INT64 relOffset = (INT64)(pActualTarget - (pSite + sizeof(INT64)));
+            INT32 relOffset = (INT32)(pActualTarget - pSite);
```

Comment:
No actually the hardware expects a PC relative encoding to be relative to the current instruction PC rather than the next instruction PC for Arm64.
Every architecture is slightly different here.